gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/GmailUtils.java

```diff
@@ -831,4 +836,103 @@ private static void addAttachmentToMessageContext(String attachmentContentID,
         axis2mc.addAttachment(attachmentContentID, handler);
         log.info("Added an attachemnt named \"" + attachmentContentID + "\" to message context");
     }
+
+
+    public static boolean validateToken (String accessToken) {
+        String validationUrl = "https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=" + accessToken;
+        try {
+            URL urlObj = new URL(validationUrl);
+            HttpsURLConnection httpsURLConnection = (HttpsURLConnection) urlObj.openConnection();
+
+            httpsURLConnection.setRequestMethod("GET");
+            if (httpsURLConnection.getResponseCode() == 200){
+                return true;
+            }
+            else {
+                return false;
+            }
+
+        } catch (MalformedURLException e) {
+            handleException("Error While Validating the Access Token", e);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
this is login. e.printStackTrace is not good idea as no can use it in. better log that in proper way